Refactoring of retractions, sections, and equivalences, and adding the 6-for-2 property of equivalences (#903)
UniMath/agda-unimath: src/foundation-core/equivalences.lagda.md

## Properties

### A map is invertible if and only if it is an equivalence

**Proof:** It is clear that if a map is
[invertible](foundation-core.invertible-maps.md), then it is also biinvertible,
and hence an equivalence.

For the converse, suppose that `f : A → B` is an equivalence with section
`g : B → A` equipped with `G : f ∘ g ~ id`, and retraction `h : B → A` equipped
with `H : h ∘ f ~ id`. We claim that the map `g : B → A` is also a retraction.

> To see this, we concatenate the following identifications
> 
> ```text
>            H (g (f x))⁻¹                  ap h (G (f x))           H x
>   g (f x) =============== h (f (g (f x)) ================ h (f x) ===== x
> ```
> 
> for any element `x : A`.

Reviewer: Maybe this proof is more conceptual by reasoning with homotopies? Anyway, I already wrote it out, so you can take it or leave it :)

````suggestion
To see this, we concatenate the following homotopies

```suggestion
         H⁻¹ ·r (g ∘ f)                 h ·l G ·r f           H
  g ∘ f ~~~~~~~~~~~~~~~ h ∘ f ∘ g ∘ f ~~~~~~~~~~~~~~~ h ∘ f ~~~~~ id.
```
````

Author: Awesome!
